### PR TITLE
Refactor Ansible playbook to extract a separate Jupyterhub role

### PR DESCRIPTION
**Pull request by asmacdo**
- [ ] Ensure that no local paths are used (see comment below)
- [ ] Get the load balancer (metallb) working: https://kind.sigs.k8s.io/docs/user/loadbalancer/ 
- [ ] Add tests to the molecule
- [ ] Add CI workflow to trigger tests
- [ ] Add basic documentation on how to use this locally: initiate section Testing with the details in README.md
- [ ] Figure out deployment aspects before merging PR:
   -  Would it break existing hub if re-deployed?

### CODE REVIEW

File: roles/jupyterhub/molecule/default/molecule.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+provisioner:
+  name: ansible
+  inventory:
+    host_vars:
+      localhost:
+        ansible_python_interpreter: '{{ ansible_playbook_python }}'
+        kubeconfig: "{{ lookup('env', 'KUBECONFIG') }}"
+  env:
+    KUBECONFIG: /home/austin/.kube/config-molecule-test
+    K8S_AUTH_KUBECONFIG: /home/austin/.kube/config-molecule-test
```
Comment (yarikoptic) [inline roles/jupyterhub/molecule/default/molecule.yml:21]: seems to have absolute paths and austin specific -- could be avoided ? may be at least via `~/` or `$HOME`?

Comment (asmacdo) [reply roles/jupyterhub/molecule/default/molecule.yml:21]: oops missed that one thanks

File: roles/jupyterhub/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+Including an example of how to use your role (for instance, with variables passed in as parameters) is always nice for users too:
+
+    - hosts: servers
+      roles:
+         - { role: username.rolename, x: 42 }
+
+License
+-------
+
+BSD
```
Comment (yarikoptic) [inline roles/jupyterhub/README.md:33]:
```suggestion
BSD-3
```
?

File: roles/jupyterhub/vars/todo.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+---
+# TODO FIX THIS HACK
```
Comment (yarikoptic) [inline roles/jupyterhub/vars/todo.yml:2]:
```suggestion
# TODO FIX THIS HACK: create proper vars definitions
# They come from https://github.com/dandi/dandi-hub/blob/dandi/group_vars/all but only the ones
# used in this playbook.
```

**Comment by asmacdo** [comment]
Paused for now

**Comment by asmacdo** [comment]
This is likely not necessary if we move forward with https://github.com/dandi/dandi-hub/pull/90

**Pull request closed without merging**